src/System.Text.Json/src/System/Text/Json/JsonReader.cs

```diff
@@ -59,7 +62,7 @@ public void Dispose()
 
         public bool Read()
         {
-            var canRead = _index < _str.Length;
```

Comment:
I am going to revert back this change. Utf8String.Length was uncharacteristically slower because the test was using debug builds. Using release builds, there isn't much difference between Utf8String.Length and me saving the property myself.